apps/prairielearn/elements/pl-number-input/pl-number-input.py

```diff
@@ -217,55 +240,37 @@ def render(element_html: str, data: pl.QuestionData) -> str:
         info_params["allow_complex"] = pl.get_boolean_attrib(
             element, "allow-complex", ALLOW_COMPLEX_DEFAULT
         )
-        info_params["show_info"] = pl.get_boolean_attrib(
-            element, "show-help-text", SHOW_HELP_TEXT_DEFAULT
-        )
+        info_params["show_info"] = show_help_text
         info_params["allow_fractions"] = allow_fractions
 
         # Find the true answer to be able to display it in the info popup
-        ans_true = None
-        if pl.get_boolean_attrib(
-            element, "show-correct-answer", SHOW_CORRECT_ANSWER_DEFAULT
-        ):
-            ans_true = format_true_ans(element, data, name)
-        if ans_true is not None:
-            info_params["a_tru"] = ans_true
+        if show_correct_answer:
+            html_params["a_tru"] = format_true_ans(element, data, name)
 
-        with open(NUMBER_INPUT_MUSTACHE_TEMPLATE_NAME, "r", encoding="utf-8") as f:
-            info = chevron.render(f, info_params).strip()
-        with open(NUMBER_INPUT_MUSTACHE_TEMPLATE_NAME, "r", encoding="utf-8") as f:
-            info_params.pop("format", None)
-            # Within mustache, the shortformat generates the placeholder that is used as a placeholder inside of the numeric entry.
-            # Here we opt to not generate the value, hence the placeholder is empty.
-            # The placeholder text may be overriden by setting the 'placeholder' attribute in the pl-number-input HTML tag
-            if pl.has_attrib(element, "placeholder"):
-                # 'placeholder' attribute is set, override the placeholder text
-                html_params["placeholder"] = pl.get_string_attrib(
-                    element, "placeholder"
-                )
-            else:
-                # 'placeholder' attribute not set, use default shortformat as placeholder text
-                info_params["shortformat"] = pl.get_boolean_attrib(
-                    element, "show-placeholder", SHOW_PLACEHOLDER_DEFAULT
-                )
-                html_params["placeholder"] = chevron.render(f, info_params).strip()
+        html_params["info"] = chevron.render(template, info_params).strip()
 
-        html_params["info"] = info
+        # Within mustache, the shortformat generates the placeholder that is used as a placeholder inside of the numeric entry.
+        # Here we opt to not generate the value, hence the placeholder is empty.
+        # The placeholder text may be overriden by setting the 'placeholder' attribute in the pl-number-input HTML tag
+        if pl.has_attrib(element, "placeholder"):
+            # 'placeholder' attribute is set, override the placeholder text
+            html_params["placeholder"] = pl.get_string_attrib(element, "placeholder")
+        else:
+            info_params.pop("format", None)
```

Comment:
Not to harp on this one line too much but: I don't understand why this is only popped when `placeholder` isn't set. When we do the final `chevron.render` for the panel, wouldn't this result in excess text being rendered?

Reply:
If a custom placeholder is set, we don't need to use the `info_params` dictionary again to generate the default placeholder. So if we are using the default, we need to remove the `format` key so the placeholder can be generated by setting the legacy `show-placeholder` key to be true on the next line.

None of this really affects the final `chevron.render`, as that is being done with `html_params` and not `info_params`.

All of that is to say, this code is using the same render logic as before, which was unfortunately super confusing.

Reply:
Ah, I missed the `info_params` vs `hhtml_params` distinction. I agree this is very confusing.